### PR TITLE
Add scarthgap compatibility

### PR DESCRIPTION
**Pull request by dsemenets**

### CODE REVIEW

**Review by lorc — CHANGES_REQUESTED** [review]
It would be really nice to have commit description.

File: meta-xt-domx/recipes-extended/xen/xen_git.bbappend [hunk 1]
```diff
@@ -1,5 +1,7 @@
+FILESEXTRAPATHS:prepend := "${THISDIR}/files:"
+
 require xen-source.inc
 
-do_configure:append() {
-    oe_runmake xt_defconfig
```
Comment (lorc) [inline meta-xt-domx/recipes-extended/xen/xen_git.bbappend:4]: Could you please describe why are you doing this change?

File: meta-xt-domx/recipes-extended/xen/files/xt-defaults.cfg [hunk 2]
```diff
@@ -0,0 +1,6 @@
+CONFIG_XSM=y
+CONFIG_TEE=y
+CONFIG_OPTEE=y
+CONFIG_IPMMU_VMSA=y
```
Comment (lorc) [inline meta-xt-domx/recipes-extended/xen/files/xt-defaults.cfg:4]: Do we really need this in common part? This is R-Car specific.

Comment (dsemenets) [reply meta-xt-domx/recipes-extended/xen/files/xt-defaults.cfg:4]: I don't know
```
do_configure:append() {
    oe_runmake xt_defconfig
```
Does the same but non-yocto way

File: meta-xt-domx/recipes-extended/xen/xen_git.bbappend (hunk 1 above)
Comment (dsemenets) [reply meta-xt-domx/recipes-extended/xen/xen_git.bbappend:4]: I described this in commit message

File: meta-xt-domx/recipes-extended/xen/files/xt-defaults.cfg (hunk 2 above)
Comment (lorc) [reply meta-xt-domx/recipes-extended/xen/files/xt-defaults.cfg:4]: You are adding a new file to the common part of build system. Prior to your change there was no config file at all. 

Probably Xen repository used by Gen3/Gen4 products include those changes, but previously they were defined in products. But this does not answer my question

File: meta-xt-domx/recipes-extended/xen/xen_git.bbappend [hunk 3]
```diff
@@ -1,5 +1,7 @@
+FILESEXTRAPATHS:prepend := "${THISDIR}/files:"
+
 require xen-source.inc
 
-do_configure:append() {
```
Comment (rshym) [inline meta-xt-domx/recipes-extended/xen/xen_git.bbappend:3]: The idea of the xt-common is that it should be compilable on any hardware platform that we support, and it should contain as generic functions as possible.

Any hardware-specific things, like `xt-defaults.cfg` added in this commit should be provided on the upper levels - xt-rpi5/xt-rcar or even in the product.

For example, if `xt-rpi5` layer has it's own `xen_git.bbappend` with it's own xt_defconfig, then `oe_runmake xt_defconfig` from this recipe will use config provided by upper level.

That's why we do not need to remove `do_configure:append()` here and do not have to add Reness-specific config inside `xt-common`. Instead, we should provide the required config in `xt-rcar` or `xt-rpi5` or `xt-prod-devl-xxx`.

Taking aside, that this commit has nothing to do with an upgrade to scarthgap.

Comment (dsemenets) [reply meta-xt-domx/recipes-extended/xen/xen_git.bbappend:3]: This is workarround and this workarround forces me do another workarround on this workarround. As I described in commit message I can't use config fragments with this recipe. If this commit out of scope this PR I will do another PR. If you don't like this solution I will offer another one. As I told in commit message - this recipe is not correct from Yokto's point of view.

File: meta-xt-domx/recipes-core/dbus/dbus-cxx_0.10.0.bb [hunk 4]
```diff
@@ -6,7 +6,7 @@ DEPENDS = "boost dbus libsigc++-2.0"
 
 LIC_FILES_CHKSUM = "file://COPYING;md5=d32239bcb673463ab874e80d47fae504"
 
-SRC_URI = "git://github.com/dbus-cxx/${PN}.git;nobranch=1;tag=${PV};protocol=https"
+SRC_URI = "git://github.com/dbus-cxx/dbus-cxx.git;nobranch=1;tag=${PV};protocol=https"
```
[the remark this answers is not in the document]
Comment (dsemenets) [reply meta-xt-domx/recipes-core/dbus/dbus-cxx_0.10.0.bb:9]: I Use master HEAD as base. Probably this is not issue for kirkstone or any previous Yocto releases but this is issue on scarthgap.

[the remark this answers is not in the document]
Comment (rshym) [reply meta-xt-domx/recipes-core/dbus/dbus-cxx_0.10.0.bb:9]: Here is the same line on master HEAD. It has no `tag=${PV};`:
https://github.com/xen-troops/meta-xt-common/blob/master/meta-xt-domx/recipes-core/dbus/dbus-cxx_0.10.0.bb#L9

Also, I have no objection to using `dbus-cxx`, and even more, I agree with that approach, and consider it better than PN or BPN. It was just a side note.

[the remark this answers is not in the document]
Comment (dsemenets) [reply meta-xt-domx/recipes-core/dbus/dbus-cxx_0.10.0.bb]: I have rebased this PR to master HEAD

File: meta-xt-domx/recipes-extended/xen/files/xt-defaults.cfg (hunk 2 above)
Comment (dsemenets) [reply meta-xt-domx/recipes-extended/xen/files/xt-defaults.cfg:4]: What this do_configure:append() does in common part? Probably this recipe should be moved to rcar layers?

**Comment by rshym** [comment]
I would propose to do two things...

1. Move fixes xen's do_configure into separate PR, which needs more investigation and discussion. At first, we need to look into full do_configure after parsing of all appends, to understand do we really do double configuration. Only after that, we can discuss how to properly fix it, if needed, taking into account our multi-level architecture (common, platform, product).

2. PR with the upgrade to scarthgap will be set to [ON HOLD] status, until we will make an upgrade for xt-rcar, products, create kirkstone branches for xt-common, xt-rcar.

File: meta-xt-domx/recipes-extended/xen/files/xt-defaults.cfg (hunk 2 above)
Comment (lorc) [reply meta-xt-domx/recipes-extended/xen/files/xt-defaults.cfg:4]: Well, yes. This particular call to `xt_defconfig` should go into meta-xt-rcar

Comment (dsemenets) [reply meta-xt-domx/recipes-extended/xen/files/xt-defaults.cfg:4]: Any way we moving that to xt-rcar doesn't fix main problem of this - double configuration in do_configure step

File: meta-xt-control-domain-virtio/conf/layer.conf [hunk 5]
```diff
@@ -9,5 +9,5 @@ BBFILE_COLLECTIONS += "xt-control-domain-virtio"
 BBFILE_PATTERN_xt-control-domain-virtio := "^${LAYERDIR}/"
 BBFILE_PRIORITY_xt-control-domain-virtio = "8"
 
-LAYERSERIES_COMPAT_xt-control-domain-virtio = "kirkstone"
+LAYERSERIES_COMPAT_xt-control-domain-virtio = "scarthgap"
```
Comment (lorc) [inline meta-xt-control-domain-virtio/conf/layer.conf:12]: I believe we need both options. I.e. LAYERSERIES_COMPAT_xt-control-domain-virtio = "scarthgap kirkstone"

File: meta-xt-control-domain/conf/layer.conf [hunk 6]
```diff
@@ -9,5 +9,5 @@ BBFILE_COLLECTIONS += "xt-control-domain"
 BBFILE_PATTERN_xt-control-domain := "^${LAYERDIR}/"
 BBFILE_PRIORITY_xt-control-domain = "8"
 
-LAYERSERIES_COMPAT_xt-control-domain = "kirkstone"
+LAYERSERIES_COMPAT_xt-control-domain = "scarthgap"
```
Comment (lorc) [inline meta-xt-control-domain/conf/layer.conf:12]: I believe we need both options. I.e. LAYERSERIES_COMPAT_xt-control-domain = "scarthgap kirkstone"

File: meta-xt-dom0/conf/layer.conf [hunk 7]
```diff
@@ -9,5 +9,5 @@ BBFILE_COLLECTIONS += "xt-dom0"
 BBFILE_PATTERN_xt-dom0 := "^${LAYERDIR}/"
 BBFILE_PRIORITY_xt-dom0 = "8"
 
-LAYERSERIES_COMPAT_xt-dom0 = "kirkstone"
+LAYERSERIES_COMPAT_xt-dom0 = "scarthgap"
```
Comment (lorc) [inline meta-xt-dom0/conf/layer.conf:12]: The same about two versions

File: meta-xt-domu-virtio/conf/layer.conf [hunk 8]
```diff
@@ -9,5 +9,5 @@ BBFILE_COLLECTIONS += "xt-domu-virtio"
 BBFILE_PATTERN_xt-domu-virtio := "^${LAYERDIR}/"
 BBFILE_PRIORITY_xt-domu-virtio = "8"
 
-LAYERSERIES_COMPAT_xt-domu-virtio = "kirkstone"
+LAYERSERIES_COMPAT_xt-domu-virtio = "scarthgap"
```
Comment (lorc) [inline meta-xt-domu-virtio/conf/layer.conf:12]: The same about two versions

File: meta-xt-domu/conf/layer.conf [hunk 9]
```diff
@@ -9,5 +9,5 @@ BBFILE_COLLECTIONS += "xt-domu"
 BBFILE_PATTERN_xt-domu := "^${LAYERDIR}/"
 BBFILE_PRIORITY_xt-domu = "8"
 
-LAYERSERIES_COMPAT_xt-domu = "kirkstone"
+LAYERSERIES_COMPAT_xt-domu = "scarthgap"
```
Comment (lorc) [inline meta-xt-domu/conf/layer.conf:12]: The same about two versions

File: meta-xt-domx/conf/layer.conf [hunk 10]
```diff
@@ -9,5 +9,5 @@ BBFILE_COLLECTIONS += "xt-domx"
 BBFILE_PATTERN_xt-domx := "^${LAYERDIR}/"
 BBFILE_PRIORITY_xt-domx = "8"
 
-LAYERSERIES_COMPAT_xt-domx = "kirkstone"
+LAYERSERIES_COMPAT_xt-domx = "scarthgap"
```
Comment (lorc) [inline meta-xt-domx/conf/layer.conf:12]: The same about two versions

File: meta-xt-driver-domain-virtio/conf/layer.conf [hunk 11]
```diff
@@ -9,5 +9,5 @@ BBFILE_COLLECTIONS += "xt-driver-domain-virtio"
 BBFILE_PATTERN_xt-driver-domain-virtio := "^${LAYERDIR}/"
 BBFILE_PRIORITY_xt-driver-domain-virtio = "8"
 
-LAYERSERIES_COMPAT_xt-driver-domain-virtio = "kirkstone"
+LAYERSERIES_COMPAT_xt-driver-domain-virtio = "scarthgap"
```
Comment (lorc) [inline meta-xt-driver-domain-virtio/conf/layer.conf:12]: The same about two versions

File: meta-xt-driver-domain/conf/layer.conf [hunk 12]
```diff
@@ -9,5 +9,5 @@ BBFILE_COLLECTIONS += "xt-driver-domain"
 BBFILE_PATTERN_xt-driver-domain := "^${LAYERDIR}/"
 BBFILE_PRIORITY_xt-driver-domain = "8"
 
-LAYERSERIES_COMPAT_xt-driver-domain = "kirkstone"
+LAYERSERIES_COMPAT_xt-driver-domain = "scarthgap"
```
Comment (lorc) [inline meta-xt-driver-domain/conf/layer.conf:12]: The same about two versions

File: meta-xt-qemu/conf/layer.conf [hunk 13]
```diff
@@ -9,4 +9,4 @@ BBFILE_COLLECTIONS += "xt-qemu"
 BBFILE_PATTERN_xt-qemu := "^${LAYERDIR}/"
 BBFILE_PRIORITY_xt-qemu = "10"
 
-LAYERSERIES_COMPAT_xt-qemu = "kirkstone"
+LAYERSERIES_COMPAT_xt-qemu = "scarthgap"
```
Comment (lorc) [inline meta-xt-qemu/conf/layer.conf:12]: The same about two versions

File: meta-xt-control-domain-virtio/conf/layer.conf (hunk 5 above)
Comment (dsemenets) [reply meta-xt-control-domain-virtio/conf/layer.conf:12]: It should be checked.

**Comment by rshym** [comment]
`LAYERSERIES_COMPAT` is not upgraded to the scarthgap for the layer `meta-xt-security` in the first commit.
This layer was not present during the creation of the PR, and was added latter.

**Comment by lorc** [comment]
> PR with the upgrade to scarthgap will be set to [ON HOLD] status, until we will make an upgrade for xt-rcar, products, create kirkstone branches for xt-common, xt-rcar.

Why, though? 

We don't need to do this in one go. If meta-xt-common will be compatible with both scarthgap and kirkstone, it will be enough both for RPi project and it should not break existing products

**Comment by rshym** [comment]
> > PR with the upgrade to scarthgap will be set to [ON HOLD] status, until we will make an upgrade for xt-rcar, products, create kirkstone branches for xt-common, xt-rcar.

> Why, though?

Yes, that was my erroneous proposal.

**Comment by rshym** [comment]
One more note...
Seems we will not be able to use
```
LAYERSERIES_COMPAT_xt-driver-domain = "kirkstone scarthgap"
```
because I found a change in meta-virtualization/.../xen.inc.
In the scarthgap they replaced `RPROVIDES = virtual/xenstored` with `virtual-xenstored`.
As a result, we have broken all backends and xen-network.
So we will have to use separate branches of the xt-common for kirkstone and scarthgap, or have to find some possibility to make yocto-dependent overrides for `RRECOMENDS`  for backends and xen-network.

That's besides the need to update `LAYERSERIES_COMPAT` for the meta-xt-security layer.

**Comment by lorc** [comment]
I believe you can check `LAYERSERIES_CORENAMES` in recipe to provide proper `RRECOMENDS` value

**Comment by rshym** [comment]
1st and 2nd commits - looks good.
But as I see we lost third commit, that removes `do_configure:append()`. And we agreed that we will merge three commits.

**Review by rshym — CHANGES_REQUESTED** [review]

**Review by lorc — APPROVED** [review]
Looks like we are good to go, thanks.

With @rshym comments addressed:

`Acked-by: Volodymyr Babchuk <volodymyr_babchuk@epam.com>`

But we need to test and merge changes to xt-rcar first.

**Comment by rshym** [comment]
With addressed two comments regarding changes to linux-* appeared suddenly:

`Acked-by: Ruslan Shymkevych <ruslan_shymkevych@epam.com>`

File: meta-xt-domu/recipes-kernel/linux-libc-headers/linux-libc-headers_6.8.bb [hunk 14]
```diff
@@ -1,6 +1,6 @@
 require recipes-kernel/linux-libc-headers/linux-libc-headers.inc
 
-KERNEL_URL = "git://github.com/torvalds/linux.git"
+KERNEL_URL = "git://github.com/torvalds/linux.git;branch=master;protocol=https"
```
[the remark this answers is not in the document]
Comment (dsemenets) [reply meta-xt-domu/recipes-kernel/linux-libc-headers/linux-libc-headers_6.8.bb:3]: This change proposed by one of poky/scripts/contrib/convert* utilities

File: meta-xt-domu/recipes-kernel/linux/linux-virtio-armv8.bb [hunk 15]
```diff
@@ -13,7 +13,7 @@ python () {
 
 FILESEXTRAPATHS:prepend := "${THISDIR}/${PN}:"
 
-BSP_URL = "git://github.com/torvalds/linux.git"
+BSP_URL = "git://github.com/torvalds/linux.git;branch=master;protocol=https"
```
[the remark this answers is not in the document]
Comment (dsemenets) [reply meta-xt-domu/recipes-kernel/linux/linux-virtio-armv8.bb:16]: Fixed

File: meta-xt-domu/recipes-kernel/linux-libc-headers/linux-libc-headers_6.8.bb (hunk 14 above)
[the remark this answers is not in the document]
Comment (dsemenets) [reply meta-xt-domu/recipes-kernel/linux-libc-headers/linux-libc-headers_6.8.bb:3]: Fixed

[the remark this answers is not in the document]
Comment (rshym) [reply meta-xt-domu/recipes-kernel/linux-libc-headers/linux-libc-headers_6.8.bb]: I suspect that mentioned script didn't look a few lines below. And is not aware, that proposed changes are already in place.
So, please remove this change.

**Pull request merged**